include/constants/battle_config.h

```diff
@@ -131,6 +131,10 @@
 #define B_FLAG_INVERSE_BATTLE       0     // If this flag is set, the battle's type effectiveness are inversed. For example, fire is super effective against water.
 #define B_FLAG_FORCE_DOUBLE_WILD    0     // If this flag is set, all land and surfing wild battles will be double battles.
 
+// Var Settings
+// To use the following features in scripting, replace the 0s with the Var ID you're assigning to it. Eg: replace VAR_UNUSED_0x40F7 with VAR_TERRAIN for that feature
+#define VAR_TERRAIN                 0 // If this var has a value, assigning a STATUS_FIELD_xx_TERRAIN to it before battle causes the battle to start with that terrain active
```

Comment:
The instructions seem a bit off in enabling this feature. And I'm not sure if the config currently works as is because I get build errors if I just replace this 0 with a value or the label of a var from vars.h.  What I had to do to get it to build was replace the var completely in vars.h which basically makes this config redundant.

Also in reference to assigning the var's value via scripting, most things look good. Assigning other field status values that aren't terrain ones is handled properly so it doesn't affect the terrain that gets set. 

One interesting thing though is that assigning STATUS_TERRAIN_ANY does affect the terrain that gets set. With no thunderstorm weather, it seems to set Misty Terrain and with thunderstorm weather it seems to set Electric Terrain, but displays the text when the Electric Terrain move used instead of the usual auto-setting text. In addition the effects of the set terrain are actually Grassy Terrain's and Psychic Terrain (assuming terrains are accurately coded). 

What might be cool to do is instead support STATUS_TERRAIN_ANY or if a user is being sneaky an or'd version of the various STATUS_FIELD_X_TERRAIN constants and actually choose a random terrain to set from those given options. So for STATUS_TERRAIN_ANY it'd be from all 4, but if I set Grassy Terrain and Psychic Terrain, it would only set between those two.

Reply:
decided to not handle randomly assigning terrain if you set more than one because it feels like a personal styling choice